src/main/java/com/epam/repository/ReservationDAO.java

```diff
@@ -0,0 +1,71 @@
+package com.epam.repository;
+
+import com.epam.model.Reservation;
+import lombok.NoArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+@NoArgsConstructor
+@Repository
+public class ReservationDAO {
+    private JdbcTemplate jdbcTemplate;
+    private RowMapper reservationMapper = (rs, rowNum) -> buildReservation(rs);
+
+    @Autowired
+    public ReservationDAO(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    Reservation buildReservation(ResultSet rs) throws SQLException {
+        return Reservation.builder()
+                .id(rs.getInt("id"))
+                .client_id(rs.getInt("client_id"))
+                .tourOffer_id(rs.getInt("tourOffer_id"))
+                .numberOfPeople(rs.getInt("numberOfPeople"))
+                .status(rs.getString("status"))
+                .discount_id(rs.getInt("discount_id"))
+                .totalPrice(rs.getInt("totalPrice"))
+                .build();
+    }
+
+    public void addReservation(Reservation reservation) {
+        String sql = "INSERT INTO reservation (client_id, tourOffer_id, " +
+                "numberOfPeople, status, discount_id, totalPrice) VALUES (?,?,?,?,?,?)";
+        jdbcTemplate.update(sql, reservation.getClient_id(), reservation.getTourOffer_id(), reservation.getNumberOfPeople(),
+                reservation.getStatus(), reservation.getDiscount_id(), reservation.getTotalPrice());
+    }
+
+    public Reservation getReservationById(Integer id) {
+        String sql = "SELECT * FROM reservation WHERE id = ?";
+        return jdbcTemplate.queryForObject(sql,
+                new Object[]{id}, (rs, rowNum) -> buildReservation(rs));
+    }
+
+    public List listReservations() {
+        String sql = "SELECT * from reservation";
+        return jdbcTemplate.query(sql, reservationMapper);
+    }
+
+    public void removeReservation(Integer id) {
+        String sql = "DELETE FROM reservation WHERE id = ?";
```

Comment:
IMHO, table names and column names can be stored in some place like property file or singleton like a static final strings. In future, when we will want to rename our columns - we will fix it in one place.

Reply:
can I create another task for that?

Reply:
yes

Reply:
done